### PR TITLE
Only fetch MAD manifesto on server join

### PR DESCRIPTION
**Pull request by Alystrasz**
Previously, the verified mods manifesto was fetched on game start without checking if the verified mod feature is enabled Squirrel
-side; with this, the manifesto is only fetched when the user wants to download a mod (meaning they enabled the feature beforehand).

**(must be merged with the mods PR!)**

#### Launcher changes

* Expose `FetchModsListFromAPI` method to Squirrel VM
* Do not invoke `FetchModsListFromAPI` method on client start
* Add `ModInstallState.MANIFESTO_FETCHING` enum value to better track mod install progress

#### TODOs

- [x] Mods counterpart PR (https://github.com/R2Northstar/NorthstarMods/pull/821)

#### Testing

*Without PRs:*

1. Launch game;
2. Check the logs: they should display verified mods manifesto was fetched close to game launch;

*With PRs:*

1. Install both launcher and mods PRs;
2. Launch game, reach MP lobby and enable MAD (`allow_mod_auto_download 1`);
3. Try and connect to a server requiring verified mods (`Space battle` or `Parkour` server should be up at the time of writing);
4. Check the logs: manifesto should be fetched when trying to join the server.

##### Media

[delayed_manifesto_fetching.webm](https://github.com/user-attachments/assets/a6b24f62-3964-4251-b6a0-10b118d0614d)

(if you look closely, you can see verified mods manifesto is fetched *WHEN* joining the server, around `0:12` in the video)

### CODE REVIEW

**Review by EladNLG — APPROVED** [review]
Reviewed it all, don't see anything needing changing, not tested.

File: primedev/mods/autodownload/moddownloader.cpp [hunk 1]
```diff
@@ -614,7 +624,12 @@ void ModDownloader::DownloadMod(std::string modName, std::string modVersion)
 ON_DLL_LOAD_RELIESON("engine.dll", ModDownloader, (ConCommand), (CModule module))
 {
 	g_pModDownloader = new ModDownloader();
+}
+
+ADD_SQFUNC("void", NSFetchVerifiedModsManifesto, "", "", ScriptContext::SERVER | ScriptContext::CLIENT | ScriptContext::UI)
+{
 	g_pModDownloader->FetchModsListFromAPI();
+	return SQRESULT_NULL;
 }
```
Comment (GeckoEidechse) [inline primedev/mods/autodownload/moddownloader.cpp:633]: Note to self that this is the essential change where the fetching is changed from on DLL load to via a call from a Squirrel function.

**Review by GeckoEidechse — APPROVED** [review]
As far as I understand the code changes they seem correct to me ^^

**Review by GeckoEidechse — APPROVED** [review]
Confirmed working in testing together with https://github.com/R2Northstar/NorthstarMods/pull/821

Enabled MAD and joined a server. Observed the delayed manifesto fetching as intended.

**Pull request merged**